Core.Collectors/IO/CloudQueueMsiWrapper.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.CloudMine.Core.Telemetry;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Queue;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Microsoft.CloudMine.Core.Collectors.IO
+{
+    public class CloudQueueMsiWrapper : IQueue
+    {
+        private CloudQueue queue;
+        private readonly string queueName;
+        private readonly string storageAccountNameEnvironmentVariable;
+        private readonly ITelemetryClient telemetryClient;
+        private DateTime msiTokenExpiration = DateTime.MinValue;
+
+        public CloudQueueMsiWrapper(CloudQueue queue, string storageAccountNameEnvironmentVariable, ITelemetryClient telemetryClient)
+        {
+            this.queue = queue;
+            this.queueName = queue.Name;
+            this.storageAccountNameEnvironmentVariable = storageAccountNameEnvironmentVariable;
+            this.telemetryClient = telemetryClient;
+        }
+
+        public Task PutObjectAsJsonStringAsync(object obj)
+        {
+            string message = JsonConvert.SerializeObject(obj);
+            return this.PutMessageAsync(message);
+        }
+
+        public async Task PutMessageAsync(string message)
```

Comment:
>async [](http://example.com/codeflow?start=15&length=5)

return both async and await (at the end) since await is the last call in the method.

Reply:
nevermind, getting the queue is awaited as well.